### PR TITLE
f1viewer: init at 2.4.0

### PR DESCRIPTION
**Pull request by michzappa**
###### Motivation for this change
An application to view Formula 1 races and highlights without a browser, using VLC or other media player.

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/f1viewer/default.nix
Comment (dotlambda) [inline pkgs/applications/video/f1viewer/default.nix:17]:
```suggestion
    description = "TUI for F1TV";
```
https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

This should also be more descriptive.

File: pkgs/applications/video/f1viewer/default.nix
Comment (dotlambda) [inline pkgs/applications/video/f1viewer/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/f1viewer/default.nix
Comment (michzappa) [reply pkgs/applications/video/f1viewer/default.nix:17]: My apologies for the missing period. I had added it, not sure where it got lost. I gave that description because that is the description the author wrote for the project's repository itself and I thought it best to stick to source.

Comment (michzappa) [reply pkgs/applications/video/f1viewer/default.nix:17]: My apologies for the period, I had some confirmation bias when I read CONTRIBUTING.md and just assumed the period was desired.

Comment (michzappa) [reply pkgs/applications/video/f1viewer/default.nix:17]: The description itself is what it is because that is what the author gives in the projects repository. I thought it best to stick to source, but I could try to think something up if required.

File: maintainers/maintainer-list.nix
Comment (legendofmiracles) [inline maintainers/maintainer-list.nix:7614]: Different commit: `maintainers: add michzappa`

File: pkgs/applications/video/f1viewer/default.nix
Comment (legendofmiracles) [inline pkgs/applications/video/f1viewer/default.nix:19]:
```suggestion
    license = licenses.gpl3Plus;
```

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/f1viewer/default.nix
Comment (michzappa) [reply pkgs/applications/video/f1viewer/default.nix]: How about 'An application to view Formula 1 races and highlights without a browser, using VLC or other media player.' ?

Comment (michzappa) [reply pkgs/applications/video/f1viewer/default.nix]: I also will note that the period has been removed from the proposed changes, keeping this conversation un-resolved to reach resolution on the contents of the description itself.

File: pkgs/applications/video/f1viewer/default.nix
Comment (legendofmiracles) [inline pkgs/applications/video/f1viewer/default.nix:21]: Set by buildGoModule

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/f1viewer/default.nix
Comment (michzappa) [reply pkgs/applications/video/f1viewer/default.nix:21]: ah, the platforms are? will remove.

File: pkgs/applications/video/f1viewer/default.nix
Comment (dotlambda) [reply pkgs/applications/video/f1viewer/default.nix]: I can't find this information in the repo. I think it should be `gpl3Only`.

Comment (legendofmiracles) [reply pkgs/applications/video/f1viewer/default.nix]: Line 637:
>  This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

Comment (dotlambda) [reply pkgs/applications/video/f1viewer/default.nix]: That's just a recommendation inside the license, but the author of f1viewer didn't put such a specification anywhere.

Comment (michzappa) [reply pkgs/applications/video/f1viewer/default.nix]: I admittedly am not very sure what a “gpl3Plus” license looks like in a repo, so I went along with the suggestion. But it does look like the repo includes specifically gpl3. When I get home I’ll try to find an example of gpl3Plus to get a better idea of the difference.

Comment (dotlambda) [reply pkgs/applications/video/f1viewer/default.nix]: Developers are advised to include a sentence like the above at the top of every source code file. Only some include the part "or (at your option) any later version."

Comment (michzappa) [reply pkgs/applications/video/f1viewer/default.nix]: Right, so I see a repo such as https://github.com/robbert-vdh/diopser, packaged as `diopser` in nixpkgs, contains GPL3 license in its repo and has such a disclaimer(? wrong word) in its files. So I understand that that is gpl3Plus. In the case of f1viewer I do not see such a disclaimer in files, and the blurb at line 637 of the license is not actually part of the license itself but instead part of the recommendation to add these disclaimers(? again, there is probably a better word) to the files. So for this package I would be on board with changing the license back to gpl3Only.

Comment (dotlambda) [reply pkgs/applications/video/f1viewer/default.nix]: Exactly.

Comment (michzappa) [reply pkgs/applications/video/f1viewer/default.nix]: Done. I think that was the last of the review feedback, thank you both. Anything else need to be done before merge? There's apparently an outstanding requested change I don't know how to get rid of, but it was addressed.

**Pull request merged**
